rts/motoko-rts/src/gc/mark_compact/bitmap.rs

```diff
@@ -2,55 +2,126 @@ use crate::mem_utils::memzero;
 use crate::memory::{alloc_blob, Memory};
 use crate::types::{size_of, Blob, Bytes, Obj};
 
+/* How the Wasm-heap maps to the bitmap
+
+  +---- RTS stack ----+---- Motoko statics ----+---- Dynamic heap --~~--+ Heap limit
+                  (prefix words)                   bitmap lives here -> | BM |
+                                                                       /      \
+                                                                      /        \
+                                       each bit represents a word -> /...bits...\
+                                          in the dynamic heap
+
+## Marking with absolute addresses
+
+When marking, we pass an absolute pointer (i.e. address space relative), for speed.
+Internally the bitmap is kept in a (non-moving) blob behind the dynamic heap (DH).
+To efficiently mark the right bit in the bitmap, we maintain a pointer that points
+_before the start of the bitmap_ such that using the `/%8`-operation on the DH
+absolute word number will address the right bit:
+
+
+       +---- BITMAP_FORBIDDEN_PTR         +---- BITMAP_PTR
+       v                                  v
+       ,          (forbidden)             ,...................... bitmap ..........~~....,
+       |   heap_prefix_words / 8 bytes    |               heap_size / 32 bytes           |
+       |   get_bitmap_forbidden_size()    |                   BITMAP_SIZE                |
+       ^                                  ^                                              ^
+      /           ^^^^^^^^^^^              \                                             |
+     /            corresponds               \                 ^^^^^^^^^^^               /
+    /                                        \                corresponds              /
+   /                                          \                                       /
+  +---- Rts stack ----+---- Motoko statics ----+--------- Dynamic heap ----------~~--+
+                                               !
+                                               ! 32-byte aligned
+
+
+Debug assertions guard the forbidden bytes from access, as this area potentially
+physically overlaps with the Motoko dynamic heap.
+
+## The alignment caveat
+
+For this scheme to work, it is essential that the start of the DH is an address that
+is divisible by 32 (implying that `heap_prefix_words % 8 == 0`). Otherwise the
+`/%8`-operation on the DH's starting address will not yield the least significant bit
+in the BM, and thus the sweep operation will be off.
+
+## Example calculation
+
+Assume the DH is at 0x80000. Assume heap limit being at 0xB0000. Then the BM thus
+could be placed at 0xB0004. Since the heap_prefix_words is 0x20000,
```

Comment:
Shouldn't the bitmap start at `0xB0008`? It has two words in the header, not one.

Reply:
Done.